apps/voting/app/src/math-utils.js

```diff
@@ -38,10 +39,88 @@ export function formatNumber(num, decimals = 2, { truncate = true } = {}) {
     : formattedNumber
 }
 
-export function percentageList(values, digits = 0) {
+export function percentageList(values) {
   return scaleBNValuesSet(values).map(value => value.toNumber())
 }
 
+// Format a percentage from BN values.
+// `value` divided by `pctBase` must be in the [0, 1] range.
+export function formatBnPercentage(
+  value,
+  pctBase,
+  { digits = 2, suffix = '%' } = {}
+) {
+  const MAX_BASE_PRECISION = 10 ** 18
+  let basePrecision = 10 ** digits
+
+  // Tolerate having too many digits by correcting the value.
+  if (basePrecision > Number.MAX_SAFE_INTEGER) {
+    basePrecision = MAX_BASE_PRECISION
+  }
+
+  return (
+    formatNumber(
+      bnPercentageToNumber(value.mul(new BN(100)), pctBase, basePrecision),
+      digits
+    ) + suffix
+  )
+}
+
+// Converts a percentage expressed as a value + base into a number between 0 and 1.
+export function bnPercentageToNumber(value, base, precision = 10 ** 9) {
+  value = new BN(value)
+  base = new BN(base)
+  return (
+    parseInt(divideRoundBigInt(value.mul(new BN(precision)), base), 10) /
```

Comment:
```suggestion
    parseInt(divideRoundBigInt(new BN(value).mul(new BN(precision)), new BN(base)), 10) /
```

re-assignment is not required I think :-)